src/test/java/com/jcabi/ssh/SSHByPasswordTest.java

```diff
@@ -0,0 +1,215 @@
+ * @author Georgy Vlasov (wlasowegor@gmail.com)
+ * @version $Id$
+ * @since 1.4
+ * @checkstyle ClassDataAbstractionCoupling (500 lines)
+ */
+public final class SSHByPasswordTest {
+    /**
+     * SSH login.
+     */
+    private static final String LOGIN = "test";
+
+    /**
+     * SSH password.
+     */
+    private static final String PASSWORD = "password";
+
+    /**
+     * Checks if {@link SSHByPassword} can execute a command on an SSH server.
+     * @throws Exception If fails
+     */
+    @Test
+    public void executesCommand() throws Exception {
+        final int port = SSHByPasswordTest.port();
+        final SshServer sshd = SSHByPasswordTest.server(port);
+        sshd.setCommandFactory(new SSHByPasswordTest.EchoCommandCreator());
+        sshd.start();
+        final String cmd = "some test command";
+        final ByteArrayOutputStream output = new ByteArrayOutputStream();
+        final Shell shell = new Shell.Verbose(
+            new SSHByPassword(
+                InetAddress.getLocalHost().getHostAddress(),
+                port,
+                LOGIN,
+                PASSWORD
+            )
+        );
+        final int exit = shell.exec(
+            cmd,
+            new NullInputStream(0L),
+            output,
+            Logger.stream(Level.WARNING, true)
+        );
+        sshd.stop();
+        MatcherAssert.assertThat(exit, Matchers.equalTo(0));
+        MatcherAssert.assertThat(output.toString(), Matchers.equalTo(cmd));
+    }
+
+    /**
+     * Setup SSH server.
+     * @param port Port to listen on.
+     * @return SSH server.
+     */
+    private static SshServer server(final int port) {
+        final SshServer sshd = SshServer.setUpDefaultServer();
+        sshd.setPort(port);
+        final PasswordAuthenticator auth =
+            Mockito.mock(PasswordAuthenticator.class);
+        Mockito.when(
+            auth.authenticate(
+                Mockito.eq(LOGIN),
+                Mockito.eq(PASSWORD),
+                Mockito.any(ServerSession.class)
+            )
+        ).thenReturn(true);
+        sshd.setPasswordAuthenticator(auth);
+        sshd.setUserAuthFactories(
+            Collections.<NamedFactory<UserAuth>>singletonList(
+                new UserAuthPassword.Factory()
+            )
+        );
+        sshd.setKeyPairProvider(
+            new SimpleGeneratorHostKeyProvider(
+                new File(Files.createTempDir(), "hostkey.ser").getAbsolutePath()
+            )
+        );
+        return sshd;
+    }
+
+    /**
+     * Allocate free port.
+     * @return Found port.
+     * @throws IOException In case of error.
+     */
+    private static int port() throws IOException {
+        final ServerSocket socket = new ServerSocket(0);
+        final int port = socket.getLocalPort();
+        socket.close();
+        return port;
+    }
+
+    /**
+     * Factory for echo command.
+     */
+    private static final class EchoCommandCreator implements CommandFactory {
```

Comment:
@Suseika `EchoCommandCreator` and `EchoCommand` classes both from `SSHByPasswordTest` and `SSHTest` are the same. Therefore let's add a puzzle here to do the following:
1. Extract `EchoCommandCreator` and `EchoCommand` into the package `com.jcabi.ssh.mock` under `src/main/java`.
2. Rename to `MkCommandCreator` and `MkCommand`.
3. Add unit tests.